templates/res/js/dashboard.js

```diff
@@ -21,14 +34,68 @@ CRM.$(function($) {
     updateCharsLeft();
   });
 
+  cj('#make-post').submit(function(e) {
+    e.preventDefault();
+
+    // Begin validation
+    if (cj('#post-content').val().length === 0) {
+      showError('Post cannot be empty.');
+      return;
+    }
+
+    if (cj('#chars-left').html() < 0) {
+      showError('Please limit the post length to ' + post_char_limit + ' characters.');
+      return;
+    }
+
+    if (!cj('#post-to-facebook').is(':checked') && !cj('#post-to-twitter').is(':checked')) {
+      showError('Please check at least one social network.');
+      return;
+    }
+
+    if (xhrOn) {
+      return;
+    }
+
```

Comment:
can you move code from 41 to 57 into different function like validateFields....?